catalyst_voices/apps/voices/lib/widgets/navigation/sections_controller.dart

```diff
@@ -128,6 +143,9 @@ final class SectionsController extends ValueNotifier<SectionsControllerState> {
         unawaited(_scrollToSectionStep(newStepId));
       }
     }
+    if (value.activeSectionId != id && value.activeStepId != null) {
+      activeStepId = Optional.of(value.activeStepId!);
+    }
```

Comment:
i don't understand this bit. Can you add comment explaining what it does and which cases solves?

Reply:
It's for this functionality ` If there is an active section selection, and the user opens segments, only the segment opens, but no active section change.` from #844. for example, the Proposal summary is closed (not expanded) and the user has selected the Title from the Proposal setup, this line of code ensures that the user can expand the Proposal summary to see sections in that segment but doesn't lose focus from the Title segment. If u have a better solution for this, please suggest

Reply:
Gotcha, thanks

Reply:
@damian-molinski what about this? Is it resolved?

Reply:
@LynxLynxx yes